src/api/profiles.js

```diff
@@ -185,4 +185,48 @@ router.put("/upload/avatar", requiresAuth, upload.single("file"), async (req, re
   }
 });
 
+// @route GET /api/profiles/listPlayers
+// @desc get all players data fro leader board ranking with accordances to rank_by filter
+// @access Private
+router.get("/leaderboardPlayers/by/:rank_by", requiresAuth, async (req, res, next) => {
+  const {rank_by} = req.params;
+  let query = '';
+
+  if(!rank_by) res.status(400).json({ error: "Missing: rank_by Attribute" });
+  //to sort documents by a field's nested object, we use a `.` operator
+  //to sort by a level-1 field of a document we just pass that as an option to $sort derictive 
+  if(rank_by === 'battles' ||  rank_by === 'wins' || rank_by === 'losses' || rank_by === 'draws'){
```

Comment:
This is fine the way it is. However, since you are checking a bunch of strings here, you could shorten this to:

`if(['battles', 'wins', 'losses', 'draws'].includes(rank_by){`

Again, I'll leave that up to you.